stan/math/prim/mat/err/check_gpu.hpp

```diff
@@ -0,0 +1,272 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
+#define STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
+
+#include <stan/math/prim/mat/fun/ocl_gpu.hpp>
+#include <stan/math/prim/arr/fun/matrix_gpu.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <sstream>
+#include <string>
+
+namespace stan {
+  namespace math {
+     /**
+     * Check if the specified matrix on the GPU is square.
+     *
+     * This check allows 0x0 matrices.
+     *
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Matrix on the GPU to test
+     *
+     * @throw <code>std::invalid_argument</code> if the matrix
+     *    is not square
+     */
+    inline void
+    check_square(const std::string& function,
+                 const std::string& name,
+                 matrix_gpu& y) {
+      check_size_match(function,
+                       "Expecting a square matrix; rows of ", name, y.rows(),
+                       "columns of ", name, y.cols());
+    }
+    /**
+     * Check if the specified matrix on the GPU has NaN values
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Matrix to test
+     *
+     * @throw <code>std::domain_error</code> if
+     *    any element of the matrix is <code>NaN</code>.
+     */
+    inline void
+    check_positive_definite_gpu(const std::string& function,
```

Comment:
It seems like there might be somewhere else where all this work is done so it isn't duplicated inside `check_positive_definite`.  It can be overloaded instead of using the suffix `_gpu`.